march_gait_selection/src/march_gait_selection/PerformGaitAction.py

```diff
@@ -19,11 +19,11 @@ def __init__(self, gait_selection):
 
     def target_gait_callback(self, goal):
         rospy.logdebug('Trying to schedule subgait %s/%s', goal.name, goal.subgait_name)
-        if not self.gait_selection.validate_gait_by_name(goal.name):
+        if not self.gait_selection.validate_gait_file(goal.name):
             rospy.logerr('Gait %s is invalid', goal.name)
             self.action_server.set_aborted('Gait ' + str(goal.name) + 'is invalid')
             return False
-        subgait = self.gait_selection.get_subgait(goal.name, goal.subgait_name)
+        subgait = self.gait_selection.get_subgait_from_loaded_subgaits(goal.name, goal.subgait_name)
```

Comment:
```suggestion
        subgait = self.gait_selection.get_subgait_from_loaded_subgaits(goal.gait_name, goal.subgait_name)
```
Otherwise this errors, because subgait has no atribute name.

Reply:
If you follow the trace you eventually end up in the from_dict method in the Subgait class, which returns a subgait. So either somewhere along the way it should have been converted to a message or it should be gait_name. It also errors on this when you run it.

Reply:
Found the problem. Comment will be posted shortly.